packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/PostProcessing/GeneratedCodeWorkspace.cs

```diff
@@ -204,16 +208,16 @@ internal static async Task<GeneratedCodeWorkspace> Create()
             return new GeneratedCodeWorkspace(project);
         }
 
-        internal static async Task<Compilation?> CreatePreviousContractFromDll(string xmlDocumentationpath, string dllPath)
+        private static async Task<CSharpCompilation?> CreateLastContractFromDll(string xmlDocumentationpath, string dllPath)
         {
             var workspace = new AdhocWorkspace();
-            Project project = workspace.AddProject("PreviousContract", LanguageNames.CSharp);
+            Project project = workspace.AddProject("LastContract", LanguageNames.CSharp);
             project = project
                 .AddMetadataReferences(_assemblyMetadataReferences.Value)
                 .WithCompilationOptions(new CSharpCompilationOptions(
                     OutputKind.DynamicallyLinkedLibrary, metadataReferenceResolver: _metadataReferenceResolver.Value, nullableContextOptions: NullableContextOptions.Disable));
             project = project.AddMetadataReference(MetadataReference.CreateFromFile(dllPath, documentation: XmlDocumentationProvider.CreateFromFile(xmlDocumentationpath)));
-            return await project.GetCompilationAsync();
+            return (CSharpCompilation?)await project.GetCompilationAsync();
```

Comment:
nit: I think we do not need to change the type to `CSharpCompilation` since we are not really using anything from the csharp version particularly, and the place that is calling this is also expecting a `Compilation` only (the method below)